x/market/client/cli/txDeal.go

```diff
@@ -100,48 +106,23 @@ func CmdDeactivateDeal() *cobra.Command {
 	return cmd
 }
 
-func NewBuildCreateDealMsg(clientCtx client.Context, txf tx.Factory, fs *flag.FlagSet) (tx.Factory, sdk.Msg, error) {
-	deal, err := parseCreateDealFlags(fs)
-	if err != nil {
-		return txf, nil, fmt.Errorf("faild to parse deal: %w", err)
-	}
-
-	budget, err := sdk.ParseCoinNormalized(deal.Budget)
-	if err != nil {
-		return txf, nil, err
-	}
-
-	msg := types.NewMsgCreateDeal(
-		deal.DataSchema,
-		&budget,
-		deal.MaxNumData,
-		deal.TrustedDataValidators,
-		clientCtx.GetFromAddress().String(),
-	)
+func NewBuildCreateDealMsg(txf tx.Factory, fs *flag.FlagSet) (tx.Factory, types.MsgCreateDeal, error) {
```

Comment:
Is there a reason to use it like this because it seems like there is no need to return `tx.Factory`?

Reply:
Oh yes, in this function the ```tx.Factory``` seems to be not needed. 
By the way, there have some problem in unmarshaling the create deal json file. So I drafted this PR.